### PR TITLE
Move tracegen functionality to telemetrygen traces subcommand

### PR DESCRIPTION
**Pull request by liustanley**
**Description:** <Describe what has changed.>
- Copies tracegen functionality to an internal package in telemetrygen
- Moves telemetrygen config from internal to main package in order to import internal traces package
- Copies existing Github actions from tracegen for telemetrygen
- Marks tracegen module as deprecated

**Link to tracking Issue:** #9597 

**Testing:** <Describe what testing was performed and which tests were added.>
Tested locally and verified existing tracegen tests pass. Only thing of note is that previously we could use `-` and `--` for flags, but now only `--` works. This might be because telemetrygen is using Cobra.

### CODE REVIEW

File: cmd/tracegen/go.mod [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// Deprecated: use github.com/open-telemetry/opentelemetry-collector-contrib/cmd/telemetrygen instead
```
Comment (mx-psi) [inline cmd/tracegen/go.mod:1]: Can you also add something on the `cmd/tracegen/README.md` about this?

File: cmd/telemetrygen/config.go [hunk 2]
```diff
@@ -35,8 +39,7 @@ var tracesCmd = &cobra.Command{
 	Short:   "Simulates a client generating traces",
 	Example: "telemetrygen traces",
 	RunE: func(cmd *cobra.Command, args []string) error {
-		err := errors.New("[WIP] The 'traces' command is a work in progress, come back later")
-		return err
+		return traces.Start(tracesCfg)
```
Comment (mx-psi) [inline cmd/telemetrygen/config.go:42]: Could you document on `cmd/telemetrygen/README.md` how the traces command works now?

File: cmd/telemetrygen/internal/traces/config.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+// Flags registers config flags.
+func (c *Config) Flags(fs *pflag.FlagSet) {
+	fs.IntVar(&c.WorkerCount, "workers", 1, "Number of workers (goroutines) to run")
+	fs.IntVar(&c.NumTraces, "traces", 1, "Number of traces to generate in each worker (ignored if duration is provided")
+	fs.BoolVar(&c.PropagateContext, "marshal", false, "Whether to marshal trace context via HTTP headers")
+	fs.Int64Var(&c.Rate, "rate", 0, "Approximately how many traces per second each worker should generate. Zero means no throttling.")
+	fs.DurationVar(&c.TotalDuration, "duration", 0, "For how long to run the test")
+	fs.StringVar(&c.ServiceName, "service", "tracegen", "Service name to use")
+
+	// unfortunately, at this moment, the otel-go client doesn't support configuring OTLP via env vars
```
Comment (jpkrohling) [inline cmd/telemetrygen/internal/traces/config.go:87]: is that still the case?

File: cmd/telemetrygen/internal/traces/worker.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+			otel.GetTextMapPropagator().Inject(childCtx, header)
+
+			// simulates getting a request from a client
+			childCtx = otel.GetTextMapPropagator().Extract(childCtx, header)
+		}
+
+		_, child := tracer.Start(childCtx, "okey-dokey", trace.WithAttributes(
+			attribute.String("span.kind", "server"),
+			semconv.NetPeerIPKey.String(fakeIP),
+			semconv.PeerServiceKey.String("tracegen-client"),
```
Comment (jpkrohling) [inline cmd/telemetrygen/internal/traces/worker.go:73]:
```suggestion
			semconv.PeerServiceKey.String("telemetrygen-client"),
```

**Comment by mx-psi** [comment]
@liustanley could you have a look at the comments by Juraci and me?

**Review by TylerHelmuth — APPROVED** [review]
LGTM, excited about telemetrygen movement.

File: cmd/telemetrygen/internal/traces/traces.go [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+)
+
+func Start(cfg *Config) error {
+	logger, err := zap.NewDevelopment()
+	if err != nil {
+		fmt.Errorf("failed to obtain logger: %w", err)
+		return err
```
Comment (TylerHelmuth) [inline cmd/telemetrygen/internal/traces/traces.go:40]:
```suggestion
		return fmt.Errorf("failed to obtain logger: %w", err)
```
This line was causing lint failures

**Review by mx-psi — APPROVED** [review]
LGTM, @jpkrohling feel free to merge after you've had one last look :)

**Review by codeboten — APPROVED** [review]

**Pull request merged**